dynamics_controllers/README.md

```diff
@@ -0,0 +1,158 @@
+achieve good performances.
+
+#### Detailed Controllers Description
+
+Below, parameters relative to each controller are listed.
+Those marked as "searched for" can live in any parent namespace since the
+`NodeHandle::searchParam` method is used to locate them.
+This facilitates "sharing" common parameters such as the gravity vector, which
+should not depend on a specific controller (at least in principle!).
+
+##### KdlChainController
+
+- `sub_controller`: should contain the configuration of a controller of type
+  compatible with `EffortJointInterface`.
+- `robot_description`: should contain the URDF of the robot. Searched for.
+- `gravity`: list with three elements, representing the gravity vector in the
+  base frame of the chain. Searched for. Default: `[0,0,0]`.
+- `chain_base`: base of the chain. Searched for. Default: name of the root link
+  from the robot description.
+- `chain_tip`: tip of the chain. Searched for. Default: if a single "branch"
+  is rooted at `base_link`, `chain_tip` will correspond to the last link of the
+  chain. If at any point multiple children are found, loading will fail.
+- `gravity_compensation_only`: as discussed above. Default: `false`.
+
+##### KdlTreeController
+
+- `sub_controller`: should contain the configuration of a controller of type
+  compatible with `EffortJointInterface`.
+- `robot_description`: should contain the URDF of the robot. Searched for.
+- `gravity`: list with three elements, representing the gravity vector in the
+  base frame of the chain. Searched for. Default: `[0,0,0]`.
+- `tree_root`: base of the tree. Searched for. Default: name of the root link
+  from the robot description.
+- `gravity_compensation_only`: as discussed above. Default: `false`.
+
+
+
+### Limitations
+
+- They all assume a static base. The main implication is that you cannot
+  "serially join" controllers for different parts of the robot.
+- They require proper identification of the Dynamic Parameters.
+- The parameters must be provided in URDF format. This is a limitation since
+  the identification will often return a smaller set of regrouped parameters.
+- The sub-controller must control all joints of the internal `Chain`/`Tree`.
+
+
+
+### Possible Improvements
+
+- [ ] Compensation of external payloads, mainly in two ways:
+    - By adding a subscriber that can update the `wrenches_` members
+    - Allowing to dynamically modify the `Chain`/`Tree` instances
+- [ ] Support for moving bases. This likely requires to re-implement the
+      controllers using another library, such as pinocchio, or to expand KDL.
+      A possible workaround is to add a "floating joint" (in the form of a
+      3T3R chain) as the parent of the base.
+- [ ] Additional parameters such as joint inertia and viscous friction.
+- [ ] Allow multiple sub-controllers to manage different joints.
+
+
+### Personal todo list:
+
+- [ ] How should the license be pasted? Some headers contain only WillowGarage,
```

Comment:
For stuff you wrote from scratch, add your uni or yourself as license holder. No need to add previous contributors.

Reply:
Addressed in last commit